src/reynir_correct/checker.py

```diff
@@ -464,6 +475,7 @@ def check_with_custom_parser(
 
 def check_with_stats(
     text: str,
+    settings: Settings,
```

Comment:
Consider moving settings until after the asterisk, to make it a keyword-only parameter.